src/pages/messages/components/TicketDateIntervalFilter.tsx

```diff
@@ -0,0 +1,95 @@
+import { Box } from '@mui/material';
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router-dom';
+import { TicketSearchParam } from '../../../api/searchApi';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { formatDateStringToISO } from '../../../utils/date-helpers';
+
+const commonDatepickerProps: Partial<DatePickerProps<Date>> = {
+  format: 'dd.MM.yyyy',
+  views: ['year', 'month', 'day'],
+  disableHighlightToday: true,
+  slotProps: {
+    textField: { sx: { maxWidth: '10rem' }, size: 'small' },
+  },
+};
+
+export const TicketDateIntervalFilter = () => {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const searchParams = new URLSearchParams(history.location.search);
+  const maxDate = new Date();
+
+  const selectedDatesParam = searchParams.get(TicketSearchParam.CreatedDate);
+  const selectedDatesParamArray = selectedDatesParam ? selectedDatesParam.split(',') : [];
+
+  const [selectedFromDate, setSelectedFromDate] = useState<Date | null>(
+    selectedDatesParamArray.length ? new Date(selectedDatesParamArray.shift() || '') : null
+  );
+
+  const [selectedToDate, setSelectedToDate] = useState<Date | null>(
+    selectedDatesParamArray.length ? new Date(selectedDatesParamArray.pop() || '') : null
+  );
+
+  const onChangeFromDate = (newDate: Date | null) => {
+    setSelectedFromDate(newDate);
+
+    const selectedFromDateString = newDate ? formatDateStringToISO(newDate) : '';
+
+    const newDateParam = selectedFromDateString + (selectedToDate ? `,${formatDateStringToISO(selectedToDate)}` : '');
+
+    updateSearchParams(newDateParam);
```

Comment:
Sliter med å henge med med ternery i et strenguuttrykk her merker jeg 😅 Vil noe sånt gjøre samme nytte? (har ikke testet, men ser ut som det er det som gjøres)
```suggestion
    const toDateString = selectedToDate ? formatDateStringToISO(selectedToDate) : '';
    updateSearchParams(`${selectedFromDateString},${toDateString}`);
```

Reply:
Ble fort en liten refactor av det meste, men tror det er mer leselig nå 🤔